src/main/java/tutorspet/model/moduleclass/UniqueModuleClassList.java

```diff
@@ -127,17 +126,13 @@ public void remove(ModuleClass toRemove) throws NullPointerException, ModuleClas
     }
 
     /**
-     * Removes the specified {@code UUID} from all {@code ModuleClass}es in the class list.
+     * Removes the specified {@code Student} from all {@code ModuleClass}es in the class list.
      */
-    public void removeUuid(UUID uuidToRemove) {
-        requireNonNull(uuidToRemove);
+    public void removeUuid(Student student) {
+        requireNonNull(student);
 
         internalList.setAll(internalList.stream()
-                .map(moduleClass -> {
-                    Set<UUID> modifiedUuids = new HashSet<>(moduleClass.getStudentUuids());
-                    modifiedUuids.remove(uuidToRemove);
-                    return new ModuleClass(moduleClass.getName(), modifiedUuids, moduleClass.getLessons());
-                })
+                .map(moduleClass -> deleteStudentFromModuleClass(moduleClass, student))
```

Comment:
Perhaps we could shift the responsibility of removing the student from module class to the command instead. Since there is currently no existing dependency from model to logic, it might be best to keep it that way to remove this cyclic dependency between the packages?

One suggestion would be for the delete and clear student commands,we can retrieve the ReadOnlyTutorsPet from the model and perform the mapping function there, and then set the new TutorsPet back into the model.

@ruixuantan do you have any suggestions on how this cyclic dependency can be resolved?